reactor-core-micrometer/src/test/java/reactor/core/observability/micrometer/TimedSchedulerTest.java

```diff
@@ -0,0 +1,340 @@
+				throw new RuntimeException(e);
+			}
+		});
+		test.schedule(() -> {
+			try {
+				Thread.sleep(1000);
+			}
+			catch (InterruptedException e) {
+				throw new RuntimeException(e);
+			}
+			finally {
+				secondTaskDone.countDown();
+			}
+		});
+
+		//there might be a slight hiccup when the registry doesn't see task1 as active
+		Awaitility.await().atMost(Duration.ofSeconds(1)).untilAsserted(
+			() -> assertThat(test.activeTasks.activeTasks()).as("one active").isOne());
+		assertThat(test.pendingTasks.activeTasks()).as("one idle").isOne();
+
+		//we give it 2s, expecting that pendingTasks and activeTasks both reflect these 2 seconds (for task2 and task1 respectively)
+		Thread.sleep(2000);
+
+		assertThat(Math.round(test.pendingTasks.duration(TimeUnit.SECONDS)))
+			.as("after 1st idle totalTime SECONDS")
+			.isEqualTo(2);
+		assertThat(Math.round(test.activeTasks.duration(TimeUnit.SECONDS)))
+			.as("after 1st active totalTime SECONDS")
+			.isEqualTo(2);
+
+		// we "resume" both tasks and let them finish, at which point the LongTaskTimers will stop recording
+		firstTaskPause.countDown();
+		secondTaskDone.await(10, TimeUnit.SECONDS);
+
+		//again, there might be a slight hiccup before registry sees 2nd task as done
+		Awaitility.await().atMost(Duration.ofSeconds(1)).untilAsserted(
+			() -> assertThat(test.activeTasks.duration(TimeUnit.SECONDS))
+			.as("once 2nd done, no active timing")
+			.isEqualTo(0));
+		assertThat(test.pendingTasks.duration(TimeUnit.SECONDS))
+			.as("once 2nd done, no pending timing")
+			.isEqualTo(0);
+		assertThat(test.pendingTasks.activeTasks()).as("at end pendingTasks").isZero();
+		assertThat(test.activeTasks.activeTasks()).as("at end activeTasks").isZero();
+
+		//now we assert that the completedTasks timer reflects an history of all Runnable#run
+		assertThat(test.completedTasks.count())
+			.as("#completed")
+			.isEqualTo(2L);
+		assertThat(test.completedTasks.totalTime(TimeUnit.MILLISECONDS))
+			.as("total duration of tasks")
+			.isCloseTo(3000, Offset.offset(500d));
+	}
+
+
+	@Test
+	void schedulePeriodicallyTimesOneRunInActiveAndAllRunsInCompleted() throws InterruptedException {
+		TimedScheduler test = new TimedScheduler(Schedulers.single(), registry, "test", Tags.empty());
+
+		//schedule a periodic task for which one run takes 500ms. we cancel after 3 runs
+		CountDownLatch latch = new CountDownLatch(3);
+		Disposable d = test.schedulePeriodically(
+			() -> {
+				try {
+					Thread.sleep(500);
+				}
+				catch (InterruptedException e) {
+					throw new RuntimeException(e);
+				}
+				finally {
+					latch.countDown();
+				}
+			},
+			100, 100, TimeUnit.MILLISECONDS);
+		latch.await(10, TimeUnit.SECONDS);
+		d.dispose();
+
+		//now we assert that the completedTasks timer reflects an history of all Runnable#run
+		assertThat(test.submittedDirect.count()).as("#submittedDirect").isZero();
+		assertThat(test.submittedPeriodicallyInitial.count()).as("#submittedPeriodically").isOne();
```

Comment:
I think we also need to test the `submittedPeriodicallyIteration.count()` to be equal to 2.

Reply:
found it and fixed in the previous round 😄